openshift/backend.dc.json

```diff
@@ -149,12 +149,55 @@
                     "type": "Recreate",
                     "recreateParams": {
                         "timeoutSeconds": 180,
-                        "mid": {
-                            "failurePolicy": "Abort",
+                        "pre": {
+                            "failurePolicy": "Retry",
                             "execNewPod": {
                                 "command": [
-                                    "/bin/sh",
+                                    "/bin/bash",
                                     "-c",
+                                    "psql -qtAX -c 'select 1'"
+                                ],
+                                "containerName": "gwells-app${NAME_SUFFIX}",
+                                "env": [
+                                    {
+                                        "name": "PGDATABASE",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "gwells-pgsql${NAME_SUFFIX}",
+                                                "key": "database-name"
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "name": "PGUSER",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "gwells-pgsql${NAME_SUFFIX}",
+                                                "key": "database-user"
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "name": "PGPASSWORD",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "gwells-pgsql${NAME_SUFFIX}",
+                                                "key": "database-password"
+                                            }
+                                        }
+                                    },
+                                    {
+                                        "name": "PGHOST",
+                                        "value": "gwells-pgsql${NAME_SUFFIX}"
+                                    }
+                                ]
+                            }
+                        },
+                        "mid": {
+                            "failurePolicy": "Retry",
```

Comment:
@garywong-bc , is `post-deply.sh` idempotent? meaning, can it safely be re-run if it fails?

Reply:
yes it can be re-run.   That's why I switched to Stephen's Django management method to create admin accounts and groups; it will not fail if they already exist.